ecrtokenrefresher/pkg/secrets/registrymirror/registrymirror.go

```diff
@@ -0,0 +1,88 @@
+package registrymirror
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/aws/eks-anywhere-packages/ecrtokenrefresher/pkg/constants"
+	k8s "github.com/aws/eks-anywhere-packages/ecrtokenrefresher/pkg/kubernetes"
+	"github.com/aws/eks-anywhere-packages/ecrtokenrefresher/pkg/secrets"
+	"github.com/aws/eks-anywhere-packages/ecrtokenrefresher/pkg/secrets/common"
+	"github.com/aws/eks-anywhere-packages/ecrtokenrefresher/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	endpointEnv = "REGISTRY_MIRROR_ENDPOINT"
+	usernameEnv = "REGISTRY_MIRROR_USERNAME"
+	passwordEnv = "REGISTRY_MIRROR_PASSWORD"
+	caEnv       = "REGISTRY_MIRROR_CACERTCONTENT"
+	secretName  = "registry-mirror-cred"
+)
+
+type RegistryMirrorSecret struct {
+	secretName       string
+	defaultClientSet *kubernetes.Clientset
+	remoteClientSets secrets.RemoteClusterClientset
+}
+
+var _ secrets.Secret = (*RegistryMirrorSecret)(nil)
+
+func (mirror *RegistryMirrorSecret) Init(defaultClientSet *kubernetes.Clientset, remoteClientSets secrets.RemoteClusterClientset) error {
+	mirror.secretName = secretName
+	mirror.defaultClientSet = defaultClientSet
+	mirror.remoteClientSets = remoteClientSets
+	return nil
+}
+
+func (mirror *RegistryMirrorSecret) IsActive() bool {
+	if val, _ := os.LookupEnv(usernameEnv); val == "" {
+		return false
+	}
+	if val, _ := os.LookupEnv(passwordEnv); val == "" {
+		return false
+	}
+	return true
```

Comment:
Should check endpoint here as well or else it would continue even with no endpoint